src/lit-html.ts

```diff
@@ -220,10 +221,18 @@ export class Template {
             parent.insertBefore(document.createTextNode(strings[i]), node);
             this.parts.push(new TemplatePart('node', index++));
           }
-        } else if (!node.nodeValue!.trim()) {
-          nodesToRemove.push(node);
-          currentNode = previousNode;
-          index--;
+        } else {
+          // Strip whitespace-only nodes, only between elements, or at the
+          // beginning or end of elements.
+          const previousSibling = node.previousSibling;
+          const nextSibling = node.nextSibling;
+          if ((previousSibling === null || previousSibling.nodeType === 1)
+              && (nextSibling === null || nextSibling.nodeType === 1)
+              && !nodeValue.trim()) {
```

Comment:
I find it stylistically unexpected that there are such explicit checks for null and yet we rely on coercion in this case. No action required.

Reply:
This is a good point. Done